src/main/java/us/kbase/sdk/common/TestLocalManager.java

```diff
@@ -0,0 +1,207 @@
+	 * @return the path.
+	 */
+	public static Path getRunDockerRelative() {
+		return TEST_LOCAL.resolve(RUN_DOCKER_SH);
+	}
+	
+	/** Get the path of the test config file relative to the module root.
+	 * @return the path.
+	 */
+	public static Path getTesCfgRelative() {
+		return TEST_LOCAL.resolve(TEST_CFG);
+	}
+	
+	/** The results of ensuring the test local directory. */
+	public static class TestLocalInfo {
+		private final Path testLocalDir;
+		private final boolean createdTestCfgFile;
+		
+		private TestLocalInfo(final Path testLocalDir, final boolean createdTestCfgFile) {
+			this.testLocalDir = testLocalDir;
+			this.createdTestCfgFile = createdTestCfgFile;
+		}
+
+		/** Get the path to the test local directory.
+		 * @return the path.
+		 */
+		public Path getTestLocalDir() {
+			return testLocalDir;
+		}
+
+		/** Get the path to the test local readme file.
+		 * @return the path.
+		 */
+		public Path getReadmeFile() {
+			return testLocalDir.resolve(README);
+		}
+
+		/** Get the path to the test local run tests shell file..
+		 * @return the path.
+		 */
+		public Path getRunTestsShFile() {
+			return testLocalDir.resolve(RUN_TESTS_SH);
+		}
+
+		/** Get the path to the test local run bash shell file.
+		 * @return the path.
+		 */
+		public Path getRunBashShFile() {
+			return testLocalDir.resolve(RUN_BASH_SH);
+		}
+
+		/** Get the path to the test local run docker shell file.
+		 * @return the path.
+		 */
+		public Path getRunDockerShFile() {
+			return testLocalDir.resolve(RUN_DOCKER_SH);
+		}
+
+		/** Get the path to the test local test config file.
+		 * @return the path.
+		 */
+		public Path getTestCfgFile() {
+			return testLocalDir.resolve(TEST_CFG);
+		}
+
+		/** Get whether the test config file was created as part of ensuring the test local
+		 * directory.
+		 * @return true if the test configuration file needed to be created. This means the user
+		 * will need to edit the file to add their credentials. 
+		 */
+		public boolean isCreatedTestCfgFile() {
+			return createdTestCfgFile;
+		}
+
+	}
+	
+	/** Create and / or restore the test local directory for an SDK app.
+	 * 
+	 * Will only overwrite files if they don't already exist.
+	 * 
+	 * @param moduleDir the SDK module root directory.
+	 * @param moduleName the name of the module.
+	 * @param dataVersion the version of the module's reference data, if applicable.
+	 * If provided, the "refdata" subfolder will be created if it doesn't exist. If is doesn't
+	 * exist, the test runner shell script will be regenerated to ensure it handles reference
+	 * data correctly.
+	 * @return information about the test local directory.
+	 * @throws IOException if a file or directory cannot be created.
+	 */
+	public static TestLocalInfo ensureTestLocal(
+			final Path moduleDir,
+			final String moduleName,
+			final Optional<String> dataVersion)
+			throws IOException {
+		final Path tlDir = requireNonNull(moduleDir, "moduleDir").resolve(TEST_LOCAL)
```

Comment:
if only you were using `Dr` as the abbreviation for directory...

Reply:
Missed opportunities abound in my PRs, it's constant heartbreak over here